octobot_trading/modes/channel/abstract_mode_producer.py

```diff
@@ -330,3 +359,70 @@ def all_databases(self):
                 provider.get_symbol_db(self.trading_mode.bot_id, self.exchange_name, self.trading_mode.symbol)
                 if self.trading_mode.symbol else None,
         }
+
+    async def _wait_for_symbol_init(self, symbol, time_frame, timeout) -> bool:
+        try:
+            await asyncio.wait_for(
+                commons_tree.EventProvider.instance().get_or_create_event(
+                    self.exchange_manager.bot_id,
+                    commons_tree.get_exchange_path(
+                        self.exchange_manager.exchange_name,
+                        common_enums.InitializationEventExchangeTopics.CANDLES.value,
+                        symbol=symbol
+                    ),
+                    allow_creation=False
+                ).wait(),
+                timeout
+            )
+            return True
+        except (asyncio.TimeoutError, concurrent.futures.TimeoutError):
+            self.logger.error(f"Initialization took more than {timeout} seconds")
+        return False
+
+    async def init_user_inputs(self, should_clear_inputs):
+        if should_clear_inputs:
+            await commons_configuration.clear_user_inputs(
+                databases.RunDatabasesProvider.instance().get_run_db(self.trading_mode.bot_id)
+            )
+        await self._register_and_apply_required_user_inputs(
+            self.get_context(None, None, self.trading_mode.symbol, None, None, None, None, None, True)
+        )
+
+    async def _register_and_apply_required_user_inputs(self, context):
+        if context.exchange_manager.is_future:
+            await basic_keywords.set_leverage(context, await basic_keywords.user_select_leverage(context))
```

Comment:
?

Reply:
Sorry, my comment wasn't clear. I was wondering why this code is in "_register_and_apply_required_user_inputs". For the rest I agree with forcing the leverage value

Reply:
The idea with `_register_and_apply_required_user_inputs` is to add and apply used inputs that should always be there when running a trading mode. I think managing leverage via user inputs makes sense so I put it there. Where would you see it ?

Reply:
Oh it's only to enable the user input to set leverage. My bad, I thought it was a exchange call.

Reply:
It might end up with an exchange call if the leverage of the given contract on exchange is different (by calling trader.set_leverage())